src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,177 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.Scanner;
+
+public class Duke {
+    public static final int MAX_NUMBER_OF_TASKS = 100;
+    public static int tasksIndex = 0;
+    public static boolean isInUse = true;
+    public static Task[] tasks;
+    public static Scanner inputReader;
+
+    public static void main(String[] args) {
+        initDuke();
+        greetUser();
+        while (isInUse) {
+            String userInput = getUserInput(inputReader);
+            String[] informationNeededForPerformingUserRequest = processUserInput(userInput);
+            performUserRequest(tasks, informationNeededForPerformingUserRequest);
+        }
+    }
+
+    public static void initDuke() {
+        tasks = new Task[MAX_NUMBER_OF_TASKS];
+        inputReader = new Scanner(System.in);
+    }
+
+    private static void greetUser() {
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    private static String getUserInput(Scanner in) {
+        return in.nextLine();
+    }
+
+    private static String[] processUserInput(String userInput) throws IndexOutOfBoundsException {

```

Comment:
You can consider using classes to encapsulate the logic for parsing and commands 👍